core/src/main/java/org/kiwix/kiwixmobile/core/utils/files/FileUtils.kt

```diff
@@ -219,4 +220,34 @@ object FileUtils {
       File("$fileName.part").exists() -> "$fileName.part"
       else -> "${fileName}aa"
     }
+
+  @JvmStatic fun writeLogFile() {
+    if (isExternalStorageWritable) {
+      val appDirectory =
+        File(Environment.getExternalStorageDirectory().toString() + "/Kiwix")
+      val logFile = File(appDirectory, "logcat.txt")
+      Log.d("KIWIX", "Writing all logs into [" + logFile.path + "]")
+
+      // create a new app folder
+      if (!appDirectory.exists()) {
+        appDirectory.mkdir()
+      }
+
+      if (logFile.exists() && logFile.isFile) {
```

Comment:
I think this could be modified a bit, as if while writing a new logs, an exception is caught, then we'll lose the previous log file as well.
Can this scenario be please looked into?

Reply:
https://github.com/kiwix/kiwix-android/issues/2143 that was the purpose of this ticket, reassess and redefine when how we log and make it useful. As well as create a class to encapsulate this feature.

I would want to see a FileLogger class injected by Dagger